example/snake/model.hpp

```diff
@@ -0,0 +1,73 @@
+//
+// lager - library for functional interactive c++ programs
+// Copyright (C) 2019 Carl Bussey
+//
+// This file is part of lager.
+//
+// lager is free software: you can redistribute it and/or modify
+// it under the terms of the MIT License, as detailed in the LICENSE
+// file located at the root of this source code distribution,
+// or here: <https://github.com/arximboldi/lager/blob/master/LICENSE>
+//
+
+#pragma once
+
+#include <cstdlib>
+#include <ctime>
+#include <utility>
+#include <vector>
+
+namespace sn {
+
+using point_t = std::pair<int, int>;
+
+inline int x(point_t p) { return p.first; }
+
+inline int y(point_t p) { return p.second; }
+
+enum class direction
+{
+    left,
+    up,
+    right,
+    down
+};
+
+struct snake_t
```

Comment:
Not fan of suffixing all types with `_t`, I only use it for aliases.

Reply:
Actually just did this because gcc complained about using the same name for the variable and type. I've now renamed it to something more verbose, but I prefer to be less verbose but still understandable for the variable (i.e., `game` instead of `g`).